front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -353,10 +353,34 @@ export function AgentMessage({
 
   const buttons: React.ReactElement[] = [];
 
-  // Standard buttons (visible when not thinking and not failed)
+  const hasMultiAgents =
+    generationContext.generatingMessages.filter(
+      (m) => m.conversationId === conversationId
+    ).length > 1;
+
+  // Show stop agent button only when streaming with multiple agents
+  // (it feels distractive to show buttons while streaming so we would like to avoid as much as possible.
+  // However, when there are multiple agents there is no other way to stop only single agent so we need to show it here).
+  if (hasMultiAgents && agentMessageToRender.status === "created") {
+    buttons.push(
+      <Button
+        key="stop-msg-button"
+        label="Stop agent"
+        variant="ghost-secondary"
+        size="xs"
+        onClick={async () => {
+          await cancelMessage([message.sId]);
+        }}
+        icon={StopIcon}
+        className="text-muted-foreground"
+      />
+    );
+  }
+
+  // Show copy & feedback buttons only when streaming is done and it didn't fail
```

Comment:
I don't really see the value of showing copy & feedback button unless the streaming is done so I changed the logic (I think we can just check `agentMessageToRender` for that 🤔?)